webapp/src/Controller/API/AbstractApiController.php

```diff
@@ -79,25 +79,16 @@ protected function getContestId(Request $request): int
 
         $qb = $this->getContestQueryBuilder($request->query->getBoolean('onlyActive', false));
         $qb
-            ->andWhere(sprintf('c.%s = :cid', $this->getContestIdField()))
+            ->andWhere('c.externalid = :cid')
             ->setParameter('cid', $request->attributes->get('cid'));
 
         /** @var Contest|null $contest */
         $contest = $qb->getQuery()->getOneOrNullResult();
 
         if ($contest === null) {
-            throw new NotFoundHttpException(sprintf('Contest with ID \'%s\' not found', $request->attributes->get('cid')));
+            throw $this->createNotFoundException(sprintf('Contest with ID \'%s\' not found', $request->attributes->get('cid')));
```

Comment:
Why did this change?

Reply:
My editor suggested it, but I will remove it. We can better do this separately.